src/sdk.ts

```diff
@@ -267,20 +271,44 @@ export class CyanSDK {
     }
 
     /**
-     * Approve Payment Plan Contract to spend a specific ERC721 token on behalf of the signer.
-     * @param {string} address - The address of the ERC721 contract.
+     * Approve Payment Plan Contract to spend a specific ERC721 or ERC1155 token on behalf of the signer.
+     * @param {string} address - The address of the ERC721 or ERC1155 contract.
      * @param {string} tokenId - The ID of the token to approve.
      * @returns {Promise<void>} - A Promise that resolves when the transaction is confirmed.
      * @throws {Error} - Throws an error if the PaymentPlan contract is not found or if the transaction fails.
      */
-    public async getApproval(address: string, tokenId: string): Promise<void> {
+    public async getApproval(address: string, tokenId: string, tokenType: ItemType = ItemType.ERC721): Promise<void> {
+        const cyanWallet = await this.getCyanWallet();
+        const chainId = await this.signer.getChainId();
         const paymentPlanContract = await this.getPaymentPlanContract();
-        const erc721Contract = ERC721Factory.connect(address, this.signer);
-        const approvedTo = await erc721Contract.getApproved(tokenId);
-
-        if (approvedTo.toLowerCase() === paymentPlanContract.address.toLowerCase()) return;
-
-        const tx = await erc721Contract.approve(paymentPlanContract.address, tokenId);
+        let tx: ContractTransaction | undefined = undefined;
+        const cyanConduitAddress = this.configs.cyanConduitAddresses.filter(
```

Comment:
return map(chainId => conduitAddress) from config.

Reply:
I used the 'find', just like the other configs.